framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -835,66 +974,74 @@ private AnnotationMirror resultAnnotationHandler(
 
     public AnnotationMirror createIntValAnnotation(List<Long> intValues) {
         intValues = ValueCheckerUtils.removeDuplicates(intValues);
-        if (intValues.isEmpty() || intValues.size() > MAX_VALUES) {
+        if (intValues.isEmpty()) {
             return UNKNOWNVAL;
```

Comment:
I am also confused with this line, which is the original implementation of handling @IntVal with no possible value. 

For example, the intersection of {3, 4, 5} and {6, 8} is {} an empty set, however, when converting to the corresponding annotation, it becomes @UnknownVal, which is inconsistent with what we've discussed (if the intersection of two ranges is empty, the corresponding annotation should be @BottomVal)